ctenex/bot/bots/alpha/exchange_bot.py

```diff
@@ -0,0 +1,299 @@
+
+    async def process_orders(
+        self,
+        orders: list[OrderGetResponse],
+        session_stream: AsyncSessionStream,
+    ) -> ProcessingResult:
+        """
+        Process the orders for the given contract.
+
+        Args:
+            orders: The orders to process (sorted by `placed_at` in ascending order).
+            session_stream: A session stream to persist the state changes.
+        """
+
+        processing_result = ProcessingResult(
+            number_of_orders_processed=0,
+            last_processed_order_timestamp=None,
+        )
+
+        if not orders:
+            logger.debug(f"No orders to process for contract {self.contract_id}")
+            return processing_result
+
+        first_order_timestamp = orders[0].placed_at
+        last_order_timestamp = orders[-1].placed_at
+
+        logger.debug(
+            f"Processing {len(orders)} orders for contract {self.contract_id}. "
+            f"Batch interval: [{first_order_timestamp} - {last_order_timestamp}]"
+        )
+
+        price_moments = []
+
+        # Calculate best bid and ask
+        best_bid, best_ask = min_and_max_price_for_limit_orders(orders)
+
+        # Assume market orders have an effective price equal to the best bid or ask
+        # TODO: Check if this assumption is correct
+        for order in orders:
+            if order.type == "market":
+                order.price = best_bid if order.side == "buy" else best_ask
+
+        # Calculate volume and price based on the sample interval
+        sample_volume = sum(order.quantity for order in orders)
+        sample_price = (
+            sum(
+                order.price * order.quantity
+                for order in orders
+                if order.price is not None
+            )
+            / sample_volume
+        )
+
+        price_moments = {
+            "timestamp": first_order_timestamp,
+            "price": float(sample_price),
+            "volume": float(sample_volume),
+            "best_bid": float(best_bid),
+            "best_ask": float(best_ask),
+        }
+
+        await self.update_state(
+            session_stream=session_stream,
+            price_moments=price_moments,
+        )
+
+        processing_result.number_of_orders_processed = len(orders)
+        processing_result.last_processed_order_timestamp = last_order_timestamp
+
+        logger.info(
+            f"Processed {processing_result.number_of_orders_processed} orders for contract {self.contract_id}. "
+            f"Interval: [{first_order_timestamp} - {last_order_timestamp}]"
+        )
+
+        return processing_result
+
+    async def update_state(
+        self,
+        session_stream: AsyncSessionStream,
+        price_moments: dict,
+    ):
+        logger.info("Updating state and strategy")
+
+        # Update price moments
+        async with session_stream() as session:
+            await session.execute(
+                text(
+                    """
```

Comment:
Might be time to consider an ORM at this point...  (If this project needed to be maintained and evolved, that is)